### ISSUE TITLE
Can't update list item

### ISSUE THREAD

**Issue by wonderplayer** [pnp/pnpcore#399]
## Category

- [x] Bug

## Describe the bug

Trying to update the list item that I got by ID.
If I follow the example provided [here](https://pnp.github.io/pnpcore/using-the-sdk/listitems-intro.html#updating-list-items), then it kinda works. One item didn't get its title updated, but the next ones did, so I guess there was another error (most likely Azure Function time out).
But, when I get list item by ID and then try to update its values, I get `Exception thrown: 'PnP.Core.CsomServiceException' in System.Private.CoreLib.dll`, which I can't debug.
I've been investigating, and I get 3 additional values when I'm trying to update just title this way:
![image](https://user-images.githubusercontent.com/15668343/115147794-73ceb600-a065-11eb-8fe5-dec445abfb4b.png)
Initially thought that there was with detecting fields for an update, but that's working good.
Actual issue happens at line 543 (file Listitem.cs)
![image](https://user-images.githubusercontent.com/15668343/115147860-b8f2e800-a065-11eb-8b45-3b9e32fb8bf2.png)
It throws exception and I don't understand what's wrong.

## Steps to reproduce
Execute code:
```
IList list = await ctx.Web.Lists.GetByIdAsync(listGUID);
IListItem item = await list.Items.GetByIdAsync(itemID);
item["Title"] = "Update from PnPCore";
item.SystemUpdate();
```

## Expected behavior

Item title gets updated.

## Environment details (*development & target environment*)

- **SDK version**: 1.1.0
- **OS**: Windows 10
- **SDK used in**: C# Azure Function
- **Framework**: .NET Core v3.1
- **Browser(s)**: N/A
- **Tooling**: Visual Studio 2019 Enterprise v16.9.4
- **Additional details**: Project file:
```
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFramework>netcoreapp3.1</TargetFramework>
    <AzureFunctionsVersion>v3</AzureFunctionsVersion>
  </PropertyGroup>
  <ItemGroup>
    <PackageReference Include="Microsoft.Azure.Functions.Extensions" Version="1.1.0" />
    <PackageReference Include="Microsoft.Azure.WebJobs.Extensions.Storage" Version="3.0.10" />
    <PackageReference Include="Microsoft.Extensions.DependencyInjection" Version="5.0.1" />
    <PackageReference Include="Microsoft.NET.Sdk.Functions" Version="3.0.11" />
    <PackageReference Include="PnP.Core" Version="1.1.0" />
    <PackageReference Include="PnP.Core.Auth" Version="1.1.0" />
  </ItemGroup>
  <ItemGroup>
    <None Update="host.json">
      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
    </None>
    <None Update="local.settings.json">
      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
      <CopyToPublishDirectory>Never</CopyToPublishDirectory>
    </None>
  </ItemGroup>
</Project>
```

### Additional context

I'm trying to update field values of a page.
Using Azure App Registration.

**Comment by wonderplayer** [comment]
Noticed one interesting thing. I'm able to update the list item, when I'm getting it in the following way:
```
await list.LoadListDataAsStreamAsync(new RenderListDataOptions()
{
    ViewXml = query,
   RenderOptions = RenderListDataOptionsFlags.ListData
});
```
But when I get item by ID, I'm not able to update it. Maybe the item is missing some system properties if it is retrieved by ID?
I will compare which properties the item has and which properties are to be updated from both ways. Also, will compare with properties after adding item and updating its properties, as per documentation.

**Comment by wonderplayer** [comment]
# Updating an item after GetByIDAsync
## List
List is Site Pages library
## Item properties
Item has following properties available:
```
item.Values.Keys
Count = 37
    [0]: "FileSystemObjectType"
    [1]: "ServerRedirectedEmbedUri"
    [2]: "ServerRedirectedEmbedUrl"
    [3]: "ContentTypeId"
    [4]: "ComplianceAssetId"
    [5]: "WikiField"
    [6]: "Title"
    [7]: "CanvasContent1"
    [8]: "BannerImageUrl"
    [9]: "Description"
    [10]: "PromotedState"
    [11]: "FirstPublishedDate"
    [12]: "LayoutWebpartsContent"
    [13]: "OData__AuthorByline"
    [14]: "OData__TopicHeader"
    [15]: "OData__SPSitePageFlags"
    [16]: "OData__OriginalSourceUrl"
    [17]: "OData__OriginalSourceSiteId"
    [18]: "OData__OriginalSourceWebId"
    [19]: "OData__OriginalSourceListId"
    [20]: "OData__OriginalSourceItemId"
    [21]: "Property1"
    [22]: "TaxKeyword"
    [23]: "Property_x0020_2"
    [24]: "OData__ModernAudienceTargetUserFieldId"
    [25]: "_ModernAudienceTargetUserField"
    [26]: "Landing_x0020_site_x0020_news"
    [27]: "Site_x0020_news_x0020_flow_x0020_in_x0020_progress"
    [28]: "ID"
    [29]: "Created"
    [30]: "AuthorId"
    [31]: "Modified"
    [32]: "EditorId"
    [33]: "OData__CopySource"
    [34]: "CheckoutUserId"
    [35]: "OData__UIVersionString"
    [36]: "GUID"
```
NOTE: *Property1*, *Property_x0020_2*, *Landing_x0020_site_x0020_news* and *Site_x0020_news_x0020_flow_x0020_in_x0020_progress* are my custom properties, so ignore them.
## Properties for update
I'm trying to system update *Landing_x0020_site_x0020_news* and *Site_x0020_news_x0020_flow_x0020_in_x0020_progress* properties.
Now when I look into the SDK core for update, I see, that SDK has detected changes in following properties:
```
request.FieldsToUpdate.Select(i => i.FieldName)
{System.Linq.Enumerable.SelectListIterator<PnP.Core.Services.Core.CSOM.Utils.Model.CSOMItemField, string>}
    [0]: "BannerImageUrl"
    [1]: "OData__AuthorByline"
    [2]: "_ModernAudienceTargetUserField"
    [3]: "Landing_x0020_site_x0020_news"
    [4]: "Site_x0020_news_x0020_flow_x0020_in_x0020_progress"
```
## Result
Exception: `Exception thrown: 'PnP.Core.CsomServiceException' in System.Private.CoreLib.dll`.
List item is not updated.

# Updating an item after LoadListDataAsStramAsync
## List
List is Site Pages library
## Item properties
Item has following properties available:
```
item.Values.Keys
Count = 28
    [0]: "PermMask"
    [1]: "FSObjType"
    [2]: "HTML_x0020_File_x0020_Type"
    [3]: "UniqueId"
    [4]: "ProgId"
    [5]: "NoExecute"
    [6]: "ContentTypeId"
    [7]: "FileRef"
    [8]: "_UIVersion"
    [9]: "SMTotalSize"
    [10]: "File_x0020_Size"
    [11]: "_CommentFlags"
    [12]: "Site_x0020_news_x0020_flow_x0020_in_x0020_progress"
    [13]: "FileLeafRef"
    [14]: "Editor"
    [15]: "Modified"
    [16]: "Landing_x0020_site_x0020_news"
    [17]: "ItemChildCount"
    [18]: "FolderChildCount"
    [19]: "A2ODMountCount"
    [20]: "_StubFile"
    [21]: "_ComplianceTag"
    [22]: "owshiddenversion"
    [23]: "_SPSitePageFlags"
    [24]: "ContentVersion"
    [25]: "DocConcurrencyNumber"
    [26]: "_VirusStatus"
    [27]: "Restricted"
```
NOTE: *Landing_x0020_site_x0020_news* and *Site_x0020_news_x0020_flow_x0020_in_x0020_progress* are my custom properties, so ignore them. Am getting limited fields by utilizing CAML query.
## Properties for update
I'm trying to system update *Landing_x0020_site_x0020_news* and *Site_x0020_news_x0020_flow_x0020_in_x0020_progress* properties.
Now when I look into the SDK core for update, I see, that SDK has detected changes in following properties:
```
{System.Linq.Enumerable.SelectListIterator<PnP.Core.Services.Core.CSOM.Utils.Model.CSOMItemField, string>}
    [0]: "Editor"
    [1]: "Landing_x0020_site_x0020_news"
```
I'm setting both *Landing_x0020_site_x0020_news* and *Site_x0020_news_x0020_flow_x0020_in_x0020_progress* to **false**, but *Site_x0020_news_x0020_flow_x0020_in_x0020_progress* is already **false**, so it's good that it doens't see it.
## Result
List item is updated.

# Updating an item after AddAsync
## Code
```
IList list = ctx.Web.Lists.GetById(new Guid(""));
Dictionary<string, object> keyValuePairs = new Dictionary<string, object>
{
    {"Title", "PnPCore title" },
    {"Some_Internal_x0020_Name", "PnPCore value"}
};
IListItem item = list.Items.Add(keyValuePairs);
item["Title"] = "PnPCore title updated";
item["Some_Internal_x0020_Name"] = "PnPCore value updated";
item.SystemUpdate();
```
## List
List is a custom list
## Item properties
Item has following properties available:
```
item.Values.Keys
Count = 2
    [0]: "Title"
    [1]: "Some_Internal_x0020_Name"
```
## Properties for update
I'm trying to system update *Title* and *CF1* properties.
Now when I look into the SDK core for update, I see, that SDK has detected changes in following properties:
```
request.FieldsToUpdate.Select(i => i.FieldName)
{System.Linq.Enumerable.SelectListIterator<PnP.Core.Services.Core.CSOM.Utils.Model.CSOMItemField, string>}
    [0]: "Title"
    [1]: "Some_Internal_x0020_Name"
```
## Result
List item is updated.

# Conclusion
Seems that list item is getting updated differently for all approaches. First approach of getting item actually adds additional properties for update, so this might be an issue, but I will leave it to you :)

**Comment by jansenbe** [comment]
Thanks for the detailed investigation @wonderplayer , I'll have a look

**Comment by jansenbe** [comment]
@wonderplayer : there was an issue (#485) which resulted in too many fields "marked as changed" when they were loaded...which then resulted in additional issues when calling one of the update methods on the list item. This fix in #485 should also handle this issue I think, so would recommend trying the latest version and see if things now work. If not, please create a new issue so we can follow-up. Thx.
